test/e2e/serial/tests/workload_placement.go

```diff
@@ -413,6 +415,305 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
+			Expect(len(pods)).To(Equal(2))
+
+			updatedPod0 := pods[0]
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod0.Spec.NodeName, targetNodeName))
+			Expect(updatedPod0.Spec.NodeName).To(Equal(targetNodeName),
+				"node landed on %q instead of on %v", updatedPod0.Spec.NodeName, targetNodeName)
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", serialconfig.Config.SchedulerName))
+			schedOK, err := nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod0.Namespace, updatedPod0.Name, serialconfig.Config.SchedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod0.Namespace, updatedPod0.Name, serialconfig.Config.SchedulerName)
+
+			rl0 := e2ereslist.FromGuaranteedPod(updatedPod0)
+			klog.Infof("post-create pod resource list: spec=[%s] updated=[%s]", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers)), e2ereslist.ToString(rl0))
+
+			nrtInitial0, err := e2enrt.FindFromList(nrtInitialList.Items, updatedPod0.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtPostCreate0, err := e2enrt.FindFromList(nrtPostCreateDeploymentList.Items, updatedPod0.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod1 := pods[1]
+			By(fmt.Sprintf("checking the pod landed on the target node %q vs %q", updatedPod1.Spec.NodeName, targetNodeName))
+			Expect(updatedPod1.Spec.NodeName).To(Equal(targetNodeName),
+				"node landed on %q instead of on %v", updatedPod1.Spec.NodeName, targetNodeName)
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", serialconfig.Config.SchedulerName))
+			schedOK, err = nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod1.Namespace, updatedPod1.Name, serialconfig.Config.SchedulerName)
+
+			rl1 := e2ereslist.FromGuaranteedPod(updatedPod1)
+			klog.Infof("post-create pod resource list: spec=[%s] updated=[%s]", e2ereslist.ToString(e2ereslist.FromContainers(podSpec.Containers)), e2ereslist.ToString(rl1))
+
+			nrtInitial1, err := e2enrt.FindFromList(nrtInitialList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			nrtPostCreate1, err := e2enrt.FindFromList(nrtPostCreateDeploymentList.Items, updatedPod1.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			var checkConsumedRes checkConsumedResFunc
+			if nrtInitial0.TopologyPolicies[0] == string(nrtv1alpha1.SingleNUMANodePodLevel) {
+				// TODO: this is only partially correct. We should check with NUMA zone granularity (not with NODE granularity)
+				checkConsumedRes = e2enrt.CheckZoneConsumedResourcesAtLeast
+			} else {
+				checkConsumedRes = e2enrt.CheckNodeConsumedResourcesAtLeast
```

Comment:
just to clarify: so if we are dealing with scope=container we check the consumedres on node level?

Reply:
When we are dealing with scope=container, the containers that are part of a pod can be on any zone on that node so we have to check for resources consumed on the entire node when comparing initial and post create NRT objects. However for scope=pod, all the containers of that pod would be allocated resources from the same zone so we need to check for resources consumed at a zone level.

Hope this helps. Let me know if you need further clarification.

Reply:
got it, thanks for the explanation